src/core/Elsa.Core/Services/WorkflowHost.cs

```diff
@@ -75,8 +76,12 @@ public class WorkflowHost : IWorkflowHost
         public async Task<WorkflowExecutionContext> RunWorkflowDefinitionAsync(string workflowDefinitionId, string? activityId, object? input = default, string? correlationId = default, CancellationToken cancellationToken = default)
         {
             var workflow = await workflowRegistry.GetWorkflowAsync(workflowDefinitionId, VersionOptions.Published, cancellationToken);
+            if (workflow == null)
+            {
+                logger.LogError("Workflow definition {WorkflowInstanceId} either not registered or not published.", workflowDefinitionId);
```

Comment:
I'm wondering if we really need to add this logging statement in addition to throwing an exception? Isn't it so that unhandled exceptions get logged anyway?

Reply:
yep @sfmskywalker all unhandled exceptions should get logged anyway if it is configured with asp.net core,but this is handled exception from ELSA and user should have more visibility for which definition id it went wrong from log.

Reply:
That makes sense.